### ISSUE TITLE
File versioning on "unshared" folder

### ISSUE THREAD

**Issue by user239393** [syncthing/syncthing#8367]
Hello all, 

I would suggest to enable file versions on unshared folders. Then you could have for example a samba share that has a file versioning. I would copy files on the share with robocopy and could get back deleted or overwrittten files.

Then Syncthing would be a tool like Synology Drive.

**Comment by tomasz1986** [comment]
Versioning local files isn't really technically possible, as Syncthing knows about modified and/or deleted files only after the event has already taken place 😉. I think Syncthing would have to be hooked deeply into the filesystem and have control over file changes while they're happening to make this possible.

Is this a duplicate of https://github.com/syncthing/syncthing/issues/1992?

**Comment by imsodin** [comment]
What @tomasz1986 said and generally it doesn't sound like a use-case Synthing is intended for. If you want to discuss your use-case and how it might be adapted to work with Syncthing, please use the forum - https://forum.syncthing.net/. The forum has a lot more eyes and friendly people on it than the issue tracker.

Thanks!

**Comment by user239393** [comment]
But you can see the changes in the changelog on the lower right corner like "file xyz was deleted..".
So Syncthing could see it I guess.

**Comment by Script-Mak3r** [comment]
Syncthing is able to log that a file has been changed, yes, but importantly, _the old version is already gone_, because you just altered it to create the new version. What purpose, then, does the existing versioning system achieve? Simply put, it handles _remote_ changes, which it's able to do because the new and old versions are both present, up until the syncing magic happens and you're working with the latest version.
